controller/RestDelivery.php

```diff
@@ -223,13 +225,43 @@ public function getStatus()
                 $this->getRequestParameter(self::TASK_ID_PARAM),
                 CompileDelivery::class
             );
-
+            $remoteEnv = $this->getStatusesFromRemoteEnv($this->getRequestParameter(self::TASK_ID_PARAM));
+            $data['remote_environments'] = $remoteEnv;
             $this->returnSuccess($data);
         } catch (\Exception $e) {
             $this->returnFailure($e);
         }
     }
 
+    /**
+     * @param $taskId
+     * @return array
+     */
+    protected function getStatusesFromRemoteEnv($taskId)
+    {
+        $taskLogEntity = $this->getTaskLogEntity($taskId);
```

Comment:
Do you really need this? If you are using it for validation reason (to check the task exist or not), it is not really necessary because it is already done by `$this->getTaskLogReturnData()` in `getStatus()`.